src/app/app.module.ts

```diff
@@ -20,8 +20,127 @@ export const DEFAULT_LIBRARY_BUTTONS = {
+          ctx.execCommand('fontName', 'Arial')
+        },
+        styleClass: 'font-arial'
+      },
+      {
+        label: 'Courier',
+        title: 'Courier',
+        command: (ctx: NgxWigComponent) => {
+          ctx.execCommand('fontName', 'Courier')
+        },
+        styleClass: 'font-courier'
+      },
+      {
+        label: 'Times',
+        title: 'Times',
+        command: (ctx: NgxWigComponent) => {
+          ctx.execCommand('fontName', 'Times')
+        },
+        styleClass: 'font-times'
+      }
+    ],
+    styleClass: ''
+  },
+  specialChar: {
+    label: 'à',
+    title: 'Accented Characters',
+    children: [
+      {
+        label: 'à',
+        title: 'à',
+        command: (ctx: NgxWigComponent) => {
+          ctx.execCommand('insertText', 'à')
+        },
+        styleClass: ''
+      },
+      {
+        label: 'è',
+        title: 'è',
+        command: (ctx: NgxWigComponent) => {
+          ctx.execCommand('insertText', 'è')
+        },
+        styleClass: ''
+      },
+      {
+        label: 'é',
+        title: 'é',
+        command: (ctx: NgxWigComponent) => {
+          ctx.execCommand('insertText', 'é')
+        },
+        styleClass: ''
+      },
+      {
+        label: 'ì',
+        title: 'ì',
+        command: (ctx: NgxWigComponent) => {
+          ctx.execCommand('insertText', 'ì')
+        },
+        styleClass: ''
+      },
+      {
+        label: 'ò',
+        title: 'ò',
+        command: (ctx: NgxWigComponent) => {
+          ctx.execCommand('insertText', 'ò')
+        },
+        styleClass: ''
+      },
+      {
+        label: 'ù',
+        title: 'ù',
+        command: (ctx: NgxWigComponent) => {
+          ctx.execCommand('insertText', 'ù')
+        },
+        styleClass: ''
+      },
+    ],
+    styleClass: '',
+    visibleDropdown: false
+  },
+  removeFormatting: {
+    label: 'R',
+    title: 'Remove formatting',
+    command: 'removeFormat',
+    styleClass: '',
+  },
+  smallcaps: {
+    label: 'S',
+    title: 'Small-caps',
+    command: (ctx: NgxWigComponent) => {
+      setSmallcaps(ctx);
+    },
+    styleClass: ''
+  }
 };
 
+function setSmallcaps(ctx: NgxWigComponent) {
+  const selectionRange = window?.getSelection()?.getRangeAt(0);
+  let selection = selectionRange?.cloneContents();
+  if (selectionRange !== undefined && selection !== undefined) {
+      if (selectionRange.startOffset !== selectionRange.endOffset) {
+          if (selectionRange.commonAncestorContainer.parentElement?.className === 'smallcaps') {
+              selectionRange.commonAncestorContainer.parentElement?.remove();
+              ctx.execCommand('insertHTML', selection?.textContent ? selection.textContent : '');
+          } else {
+              let span = document.createElement('span');
+              span.appendChild(selection);
+              let wrappedselection = '<span class="smallcaps" style="font-variant: small-caps;">' + span.innerHTML + '</span>';
```

Comment:
the logic is a bit strange for me here. 

Do you create a span, put content there, take the content, and put it into another span? 

Maybe it's good to create an only wrapper and put content there?

Reply:
Since this is just a classic custom button used as an example, would it be better to remove it from the PR? This is old code, written for one of my projects, so it is not really related to this PR.